### PR TITLE
Extension Module Compat Cleanup

### PR DESCRIPTION
**Pull request by WillAyd**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/_libs/src/parser/io.c [hunk 1]
```diff
@@ -17,7 +17,7 @@ The full license is in the LICENSE file, distributed with this software.
 #define O_BINARY 0
 #endif  // O_BINARY
 
-#if PY_VERSION_HEX >= 0x03060000 && defined(_WIN32)
+#ifdef(_WIN32)
```
Comment (gfyoung) [inline pandas/_libs/src/parser/io.c]: I saw some compilation failures around this.

Can't we just do a pure clean and remove the `PY_VERSiON_HEX` from that if statement?

File: pandas/_libs/src/compat_helper.h [hunk 2]
```diff
@@ -38,13 +38,9 @@ PANDAS_INLINE int slice_get_indices(PyObject *s,
                                     Py_ssize_t *stop,
                                     Py_ssize_t *step,
                                     Py_ssize_t *slicelength) {
-#if PY_VERSION_HEX >= 0x03000000
+
   return PySlice_GetIndicesEx(s, length, start, stop,
                               step, slicelength);
```
Comment (jbrockmendel) [inline pandas/_libs/src/compat_helper.h]: if we're returning this unchanged, can we just use this function directly and not define an alias?  i guess the pypy thing above might complicate this?

Comment (gfyoung) [reply pandas/_libs/src/compat_helper.h]: Unclear at this point, but makes sense to investigate in a follow-up.

**Comment by jreback** [comment]
lgtm, pending green.

**Comment by jreback** [comment]
thanks

**Pull request merged**